### PR TITLE
Add quality metrics to the nit catalog

### PR DESCRIPTION
**Pull request by privat**
Three new metrics:

* number of errors (boo)
* number of warnings (and advices)
* percentage of documentation (with ponderation)

The 3 firsts commits was to make errors&warnings accessible without printing them.

### CODE REVIEW

**Comment by privat** [comment]
update: forgot tests.

By the way, since advices are now always put in the AST, then nitlight and highlight do show them in their output as some (ugly) pop-up links. Is there anyone that use nitlight or highlight? What do you prefer?

File: src/toolcontext.nit [hunk 1]
```diff
@@ -58,6 +58,13 @@ class Message
 	# * enclose identifiers, keywords and pieces of code with back-quotes.
 	var text: String
 
+	# The severity level
+	#
+	# * 0 is advices (see `ToolContext::advice`)
+	# * 1 is warnings (see `ToolContext::warning`)
+	# * 2 is errors (see `ToolContext::error`)
+	var level: Int
```
Comment (jcbrinfo) [inline src/toolcontext.nit]: I’m not a fan of magic constants. Simply adding a new severity level would be painful.

Comment (privat) [inline src/toolcontext.nit]: I agree. I will change them once we have enumerations :)

Comment (jcbrinfo) [inline src/toolcontext.nit]: For now, I think you should do it like with any language without enums: use symbolic constants. :)

**Comment by Morriar** [comment]
+1

**Comment by R4PaSs** [comment]
:shipit:

**Pull request merged**